### PR TITLE
226 c4.0

### PR DESCRIPTION
**Pull request by paulchandler**
Add audit logs for prepare statements.
#226

### CODE REVIEW

File: ecaudit/src/main/java/com/ericsson/bss/cassandra/ecaudit/AuditAdapter.java [hunk 1]
```diff
@@ -107,7 +108,30 @@ public void auditRegular(String operation, ClientState state, Status status, lon
             auditor.audit(logEntry);
         }
     }
+    /**
```
Comment (tommystendahl) [inline ecaudit/src/main/java/com/ericsson/bss/cassandra/ecaudit/AuditAdapter.java:111]: Add empty line above

File: ecaudit/src/main/java/com/ericsson/bss/cassandra/ecaudit/config/AuditConfig.java [hunk 2]
```diff
@@ -119,6 +119,11 @@ public void setWhitelistCacheMaxEntries(int whitelistCacheMaxEntries)
         loadConfigIfNeeded();
         yamlConfig.setWhitelistCacheMaxEntries(whitelistCacheMaxEntries);
     }
+    public boolean isSuppressPrepareStatements()
```
Comment (tommystendahl) [inline ecaudit/src/main/java/com/ericsson/bss/cassandra/ecaudit/config/AuditConfig.java:122]: Add empty line above

File: ecaudit/src/main/java/com/ericsson/bss/cassandra/ecaudit/config/AuditYamlConfig.java [hunk 3]
```diff
@@ -157,4 +158,10 @@ public void setWhitelistCacheMaxEntries(Integer whitelistCacheMaxEntries)
     {
         this.whitelist_cache_max_entries = whitelistCacheMaxEntries;
     }
-}
+    public Boolean isSuppressPrepareStatements()
```
Comment (tommystendahl) [inline ecaudit/src/main/java/com/ericsson/bss/cassandra/ecaudit/config/AuditYamlConfig.java:161]: Add empty line above

File: ecaudit/src/main/java/com/ericsson/bss/cassandra/ecaudit/entry/suppressor/PrepareAuditOperation.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    {
+        this.operationString = "Prepared: " + operationString;
+    }
+
+    @Override
+    public String getOperationString()
+    {
+        return operationString;
+    }
+    @Override
```
Comment (tommystendahl) [inline ecaudit/src/main/java/com/ericsson/bss/cassandra/ecaudit/entry/suppressor/PrepareAuditOperation.java:38]: Add empty line above

File: ecaudit/src/main/java/com/ericsson/bss/cassandra/ecaudit/filter/role/RoleAuditFilter.java [hunk 5]
```diff
@@ -74,6 +74,10 @@ public void setup()
         whitelistDataAccess.setup();
     }
 
+    @Override
+    public boolean shouldLogPrepareStatements() { return true; }
+
+
```
Comment (tommystendahl) [inline ecaudit/src/main/java/com/ericsson/bss/cassandra/ecaudit/filter/role/RoleAuditFilter.java:80]: remove one empty line

File: ecaudit/src/main/java/com/ericsson/bss/cassandra/ecaudit/filter/yamlandrole/YamlAndRoleAuditFilter.java [hunk 6]
```diff
@@ -57,4 +57,10 @@ public void setup()
         yamlFilter.setup();
         roleFilter.setup();
     }
+    @Override
+    public boolean shouldLogPrepareStatements()
+    {
+        return yamlFilter.shouldLogPrepareStatements();
+    }
+
```
Comment (tommystendahl) [inline ecaudit/src/main/java/com/ericsson/bss/cassandra/ecaudit/filter/yamlandrole/YamlAndRoleAuditFilter.java:65]: remove empty line

File: ecaudit/src/test/java/com/ericsson/bss/cassandra/ecaudit/handler/TestAuditQueryHandler.java [hunk 7]
```diff
@@ -146,6 +146,10 @@ public void testPrepareAndGetPrepared()
 
         verify(mockHandler, times(1)).prepare(eq(query), eq(mockClientState), eq(customPayload));
         verify(mockHandler, times(1)).getPrepared(eq(statementId));
+        verify(mockAdapter, times(1)).auditPrepare(eq(query), eq(mockClientState), eq(Status.ATTEMPT), longThat(isCloseToNow()));
+
```
Comment (tommystendahl) [inline ecaudit/src/test/java/com/ericsson/bss/cassandra/ecaudit/handler/TestAuditQueryHandler.java:150]: remove empty lines

File: integration-test-query-logger/src/test/java/com/ericsson/bss/cassandra/ecaudit/integration/querylogger/ITQueryLogger.java [hunk 8]
```diff
@@ -94,7 +98,33 @@ public void testBasicStatement()
...
+    {
+        givenTable("school", "students");
+        reset(mockAuditAppender);
 
+        PreparedStatement prepared = session.prepare("INSERT INTO school.students (key, value) VALUES (?, ?)");
+        session.execute(prepared.bind(42, "Kalle"));
+        assertThat(getLogEntries()).containsOnly( "client:'127.0.0.1'|user:'anonymous'|status:'ATTEMPT'|operation:'Prepared: INSERT INTO school.students (key, value) VALUES (?, ?)'",
+                                                  "client:'127.0.0.1'|user:'anonymous'|status:'ATTEMPT'|operation:'INSERT INTO school.students (key, value) VALUES (?, ?)[42, 'Kalle']'");
+
+//      assertThat(getLogEntries()).containsOnly("client:'127.0.0.1'|user:'anonymous'|status:'ATTEMPT'|operation:'INSERT INTO school.students (key, value) VALUES (?, ?)[42, 'Kalle']'");
```
Comment (tommystendahl) [inline integration-test-query-logger/src/test/java/com/ericsson/bss/cassandra/ecaudit/integration/querylogger/ITQueryLogger.java:112]: remove commented code

File: integration-test-query-logger/src/test/java/com/ericsson/bss/cassandra/ecaudit/integration/querylogger/ITQueryLogger.java [hunk 9]
```diff
@@ -94,7 +98,33 @@ public void testBasicStatement()
...
+    {
+        givenTable("school", "students");
+        reset(mockAuditAppender);
+
+        assertThatExceptionOfType(InvalidQueryException.class).isThrownBy(() ->  session.prepare("INSERT INTO school.invalidestudents (key, value) VALUES (?, ?)"));
+
+        assertThat(getLogEntries()).containsOnly( "client:'127.0.0.1'|user:'anonymous'|status:'ATTEMPT'|operation:'Prepared: INSERT INTO school.invalidestudents (key, value) VALUES (?, ?)'",
+                                                  "client:'127.0.0.1'|user:'anonymous'|status:'FAILED'|operation:'Prepared: INSERT INTO school.invalidestudents (key, value) VALUES (?, ?)'");
+
+//        verify(mockAuditAppender, never() ).doAppend(loggingEventCaptor.capture());
```
Comment (tommystendahl) [inline integration-test-query-logger/src/test/java/com/ericsson/bss/cassandra/ecaudit/integration/querylogger/ITQueryLogger.java:125]: remove commented code

**Review by tommystendahl — CHANGES_REQUESTED** [review]

**Comment by paulchandler** [comment]
All items fixed

**Review by tommystendahl — APPROVED** [review]

**Pull request merged**